From 9cdae6942881670af2049a51e24aa38236af7e36 Mon Sep 17 00:00:00 2001
From: Jonathon Penix <jpenix@quicinc.com>
Date: Thu, 20 Nov 2025 15:46:48 -0800
Subject: [PATCH] Experimentally try to add llvm install targets for eld

The idea here is that it will make eld work better with
LLVM_DISTRIBUTION_COMPONENTS, which apparently looks for these install
targets.

Signed-off-by: Jonathon Penix <jpenix@quicinc.com>
---
 tools/eld/CMakeLists.txt | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/tools/eld/CMakeLists.txt b/tools/eld/CMakeLists.txt
index 208918af..06d37192 100644
--- a/tools/eld/CMakeLists.txt
+++ b/tools/eld/CMakeLists.txt
@@ -8,6 +8,10 @@ set(LLVM_LINK_COMPONENTS ${LLVM_TARGETS_TO_BUILD} irreader Support)
 add_llvm_executable(ld.eld eld.cpp)
 add_dependencies(ld.eld update-eld-verinfo)
 
+add_llvm_install_targets(install-ld.eld
+                         DEPENDS ld.eld
+                         COMPONENT ld.eld)
+
 set(LINKER_WRAPPER_LIB "LW")
 target_link_libraries(ld.eld PRIVATE ${LINKER_WRAPPER_LIB})
 
-- 
2.17.1

